sky/utils/schemas.py

```diff
@@ -590,7 +592,29 @@ def get_config_schema():
             'additionalProperties': False,
             'properties': {
                 'security_group_name': {
-                    'type': 'string'
+                    'oneOf': [
+                        {
+                            'type': 'string'
+                        },
+                        {
+                            # A list of single-element dict to pretain the
+                            # order.
+                            # Example:
+                            #  security_group_name:
+                            #    - my-cluster1-*: my-security-group-1
+                            #    - my-cluster2-*: my-security-group-2
+                            #    - "*"": my-security-group-3
+                            'type': 'array',
+                            'items': {
+                                'type': 'object',
+                                'additionalProperties': {
+                                    'type': 'string'
+                                },
+                                'maxProperties': 1,
+                                'minProperties': 1,
+                            },
+                        }
```

Comment:
minor nit: Since this pattern has been used by the `remote_identity` and `security_group_name`, we can probably have a constant or a function to generate this dict.